app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -53,36 +53,61 @@ public TrainingHelper(Context context){
     }
 
 
+    /**
+     * Get all the StudentImages where the features haven't been extracted yet
+     * Extract features for every StudentImage and store them as StudentImageFeature
+     */
     public void extractFeatures(){
-        // Get all the StudentImages where the features haven't been extracted yet
         List<StudentImage> studentImageList = studentImageDao.queryBuilder()
                 .where(StudentImageDao.Properties.StudentImageFeatureId.eq(0))
                 .list();
         Log.d(getClass().getName(), "Number of StudentImages, where the features haven't been extracted yet: " + studentImageList.size());
         TensorFlow tensorFlow = getInitializedTensorFlow();
         for(StudentImage studentImage : studentImageList){
             String svmVector = getSvmVector(tensorFlow, studentImage);
-            storeStudentImageFeature(studentImage, svmVector);
+            if (svmVector != null){
+                storeStudentImageFeature(studentImage, svmVector);
+            }
+            // TODO housekeeping job #226
         }
     }
 
+    /**
+     * Stores a StudentImageFeature to the database
+     * @param studentImage - StudentImage
+     * @param svmVector - Extracted features converted to an SVM string for LIBSVM (without the label)
+     */
     private void storeStudentImageFeature(StudentImage studentImage, String svmVector){
         StudentImageFeature studentImageFeature = new StudentImageFeature(studentImage.getId(), Calendar.getInstance(), svmVector);
         studentImage.setStudentImageFeature(studentImageFeature);
         studentImageFeatureDao.insert(studentImageFeature);
         studentImageDao.update(studentImage);
-        Log.d(getClass().getName(), "StudentImageFeature for StudentImage with Id " + studentImage.getId() + " has been extracted and stored.");
+        Log.d(getClass().getName(), "StudentImageFeature with Id " + studentImageFeature.getId() + " for StudentImage with Id " + studentImage.getId() + " has been extracted and stored.");
     }
 
+    /**
+     * Load image into OpenCV Mat object
+     * Extract features from TensorFlow model
+     * Convert feature to SVM string
+     * @param tensorFlow
+     * @param studentImage
+     * @return
```

Comment:
see https://drive.google.com/a/literacyapp.org/file/d/0B3jQsJcchixPT085bmh4TDBsamc/view?usp=sharing (pulled SqLite db from device...in STUDENT_IMAGE_FEATURE you can see the typical sizes of the SVM string which is about 40'000 characters)